src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -1,24 +1,232 @@
+    setAuthState(AUTH_AUTHENTICATING)
+    isAuthTokenValid(token)
+      .then(v => {
+        setAuthState(AUTH_AUTHENTICATED)
+        return v
+      })
+      .catch(e => {
+        if (e instanceof TypeError) {
+          // network/service error
+          setAuthState(AUTH_SERVICE_UNAVAILABLE)
+        } else {
+          setAuthState(AUTH_AUTHENTICATION_FAILED)
+        }
+      })
+  }, [setAuthState, token])
+
+  useEffect(() => {
+    if (!email && !token) {
+      setAuthState(AUTH_UNAUTHENTICATED)
+      return
+    }
+
+    if (email && !token) {
+      setAuthState(AUTH_PREVERIFY)
+      return
+    }
+
+    if (email && token) {
+      handleAuthenticate()
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [email, token])
+
+  useEffect(() => {
+    token
+      ? localStorage.setItem(NOTIFICATION_SERVICE_TOKEN_KEY, token)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_TOKEN_KEY)
+  }, [token])
+
+  useEffect(() => {
+    email
+      ? localStorage.setItem(NOTIFICATION_SERVICE_EMAIL_KEY, email)
+      : localStorage.removeItem(NOTIFICATION_SERVICE_EMAIL_KEY)
+  }, [email])
+
+  const handleLogout = useCallback(() => {
+    setEmail(null)
+    setToken(null)
+  }, [setToken, setEmail])
+
+  const setServiceUnavailable = useCallback(() => {
+    setAuthState(AUTH_SERVICE_UNAVAILABLE)
+  }, [setAuthState])
+
+  return {
+    authState,
+    email,
+    token,
+    handleTokenChange: setToken,
+    handleEmailChange: setEmail,
+    handleLogout,
+    setServiceUnavailable,
+    handleAuthenticate,
+  }
 }
 
-export default Notifications
+export default function Notifications({
+  apps,
+  subsection,
+  dao,
+  handleNavigation,
+  navigationIndex,
+}) {
+  const {
+    authState,
+    email,
+    token,
+    handleAuthenticate,
+    handleTokenChange,
+    handleEmailChange,
+    handleLogout,
+    setServiceUnavailable,
+  } = useAuthState()
+
+  const navigateToNotifications = useCallback(
+    () => handleNavigation(navigationIndex),
+    [handleNavigation, navigationIndex]
+  )
+
+  if (subsection && subsection.startsWith(VERIFY_SUBSECTION)) {
+    return (
+      <NotificationsVerify
+        subsection={subsection}
+        onTokenChange={handleTokenChange}
+        onEmailChange={handleEmailChange}
+        navigateToNotifications={navigateToNotifications}
+      />
+    )
+  }
+
+  switch (authState) {
+    case AUTH_PREVERIFY:
+      return (
+        <NotificationsPreVerify
+          email={email}
+          onEmailChange={handleEmailChange}
+        />
+      )
+    case AUTH_AUTHENTICATED:
+      return (
+        <ManageNotifications
+          onServiceUnavailable={setServiceUnavailable}
+          dao={dao}
+          onLogout={handleLogout}
+          apps={apps}
+          email={email}
+          token={token}
+        />
+      )
+    case AUTH_AUTHENTICATING:
+      // return null to prevent flickering
+      return null
+    case AUTH_AUTHENTICATION_FAILED:
```

Comment:
![image](https://user-images.githubusercontent.com/3072458/64101993-4ee1de00-cd6f-11e9-93ba-13bb801b2885.png)

It feels like too much spacing here but it may be me, let's just double check the designs here.

Reply:
Yep. It's 12px in the designs. So `1.5 * GU` instead of `2 * GU`.